sdk/python/lib/pulumi/x/automation/_stack.py

```diff
@@ -352,19 +339,39 @@ def on_exit_fn():
             args.append(f"--client=127.0.0.1:{port}")
         args.extend(["--exec-kind", kind])
 
+        log_file, temp_dir = _create_log_file("preview")
+        args.extend(["--event-log", log_file])
+        summary_events: List[SummaryEvent] = []
+
+        def on_event_callback(event: EngineEvent):
```

Comment:
Since it does not have `-> <return-type>` annotation, mypy skips checking the body of this def. Just FYI.